src/Persistence_representations/include/gudhi/PSSK.h

```diff
@@ -5,163 +5,167 @@
+ *
+ * @ingroup Persistence_representations
  **/
 class PSSK : public Persistence_heat_maps<constant_scaling_function>
 {
  public:
-  PSSK() : Persistence_heat_maps() {}
+  using Base = Persistence_heat_maps<constant_scaling_function>;
+
+  PSSK() : Base() {}
 
   PSSK(const std::vector<std::pair<double, double> >& interval,
-       std::vector<std::vector<double> > filter = create_Gaussian_filter(5, 1),
-       size_t number_of_pixels = 1000,
-       double min_ = -1,
-       double max_ = -1)
-      : Persistence_heat_maps()
+       const std::vector<std::vector<double> >& filter = create_Gaussian_filter(5, 1),
+       std::size_t number_of_pixels = 1000,
+       double min = -1,
+       double max = -1)
+      : Base()
   {
-    this->construct(interval, filter, number_of_pixels, min_, max_);
+    this->_construct(interval, filter, number_of_pixels, min, max);
   }
 
   PSSK(const char* filename,
-       std::vector<std::vector<double> > filter = create_Gaussian_filter(5, 1),
-       size_t number_of_pixels = 1000,
-       double min_ = -1,
-       double max_ = -1,
-       unsigned dimension = std::numeric_limits<unsigned>::max())
-      : Persistence_heat_maps()
+       const std::vector<std::vector<double> >& filter = create_Gaussian_filter(5, 1),
+       std::size_t number_of_pixels = 1000,
+       double min = -1,
+       double max = -1,
+       unsigned int dimension = std::numeric_limits<unsigned int>::max())
+      : Base()
   {
-    std::vector<std::pair<double, double> > intervals_;
-    if (dimension == std::numeric_limits<unsigned>::max()) {
-      intervals_ = read_persistence_intervals_in_one_dimension_from_file(filename);
+    std::vector<std::pair<double, double> > intervals;
+    if (dimension == std::numeric_limits<unsigned int>::max()) {
+      intervals = read_persistence_intervals_in_one_dimension_from_file(filename);
     } else {
-      intervals_ = read_persistence_intervals_in_one_dimension_from_file(filename, dimension);
+      intervals = read_persistence_intervals_in_one_dimension_from_file(filename, dimension);
     }
-    this->construct(intervals_, filter, number_of_pixels, min_, max_);
+    this->_construct(intervals, filter, number_of_pixels, min, max);
   }
 
- protected:
-  void construct(const std::vector<std::pair<double, double> >& intervals_,
-                 std::vector<std::vector<double> > filter = create_Gaussian_filter(5, 1),
-                 size_t number_of_pixels = 1000,
-                 double min_ = -1,
-                 double max_ = -1);
-};
-
-// if min_ == max_, then the program is requested to set up the values itself based on persistence intervals
-void PSSK::construct(const std::vector<std::pair<double, double> >& intervals_,
-                     std::vector<std::vector<double> > filter,
-                     size_t number_of_pixels,
-                     double min_,
-                     double max_)
-{
-  bool dbg = false;
```

Comment:
IIRC Paweł did this on purpose, because he wanted to be able to turn these on individually (not all or nothing), at runtime, in the debugger. That doesn't mean we have to keep it, but I thought I'd post this for context.

Reply:
Yes, Pawel did this on purpose, but I also prefer the debug traces mechanism that was put in places for all the other modules.

Reply:
I just found out that there is a macro command `#undef`. I guess, someone could play with it to stop other methods from printing if one only wants specific outputs. In the old version, he had to re-compile every time too. The only annoying part, is that we have to modify not only the method we want to print. Or we could add the possibility to compile gudhi in debug mode, but without defining `GUDHI_DEBUG`, so the user can define and undefine it by hand in the place he wants.

Reply:
As the output where full of `getchar` (and another similar one with similar effect I don't remember the name), I do not think it was intended to be used with a debugger. That is why, I thought about the more classic version of debugging "by hand" with printings. Also, if you use a debugger, are the traces even useful? The debugger already gives you the values of most objects. In that case, I would simply prefer to have `GUDHI_DEBUG` not defined.